core-skills/web-application-development-with-react/testing/components.md

```diff
@@ -0,0 +1,142 @@
+# Unit Testing React Components
+
+## Introduction
+
+One of the benefits of using React over more traditional server rendered approaches
+is the ability to unit test each component in isolation. However, it is sometimes not entirely clear
+how to best approach unit testing a React component.
+
+## Things to consider
+
+### What to test
+
+Much like when unit testing a class, you want to be testing the observable outputs of running something. For example, a use case which calls a gateway. You will test the gateway was called, the use case returned
+what you expected, but not the internal state during execution.
+
+Similarly, with a React component it can be tempting to test things such as the internal state, but
+you instead want to test the observable outputs. This is most likely going to be something displaying
+(or not displaying) on the page as a result of various props, or user interactions. You are also most
+likely going to want to test that something happens when components are mounted, or that they can call
+various props given to them.
+
+### What not to test
+
+Frontend tests can be brittle if you test the wrong things, for example, what a header says. If this is
+static text that isn't tied to any behaviour you shouldn't test this, as it could be changed and cause
+test failures.
+
+### Enzyme - Mount vs Shallow
+
+<< mount vs shallow, mount is more consistent as shallow is similar to mocking classes you don't own >>
```

Comment:
No sure I agree with this. Where possible, we should be testing against shallow as mount will render all of the child components which means you're already not testing a component in isolation.
If there's something in the child component that needs to be tested, it should be done in the child component's tests.